rust/datafusion/src/datasource/parquet.rs

```diff
@@ -262,20 +263,55 @@ impl ParquetFile {
                                 is_nullable,
                             )?
                         }
-                        ColumnReader::Int32ColumnReader(ref mut r) => {
-                            ArrowReader::<Int32Type>::read(
+                        ColumnReader::Int32ColumnReader(ref mut r) => match dt {
+                            DataType::Date32(DateUnit::Millisecond) => {
+                                ArrowReader::<Date32MillisecondType>::read(
+                                    r,
+                                    self.batch_size,
+                                    is_nullable,
+                                )?
+                            }
+                            DataType::Time32(TimeUnit::Millisecond) => {
+                                ArrowReader::<Time32MillisecondType>::read(
+                                    r,
+                                    self.batch_size,
+                                    is_nullable,
+                                )?
+                            }
+                            _ => ArrowReader::<Int32Type>::read(
                                 r,
                                 self.batch_size,
                                 is_nullable,
-                            )?
-                        }
-                        ColumnReader::Int64ColumnReader(ref mut r) => {
-                            ArrowReader::<Int64Type>::read(
+                            )?,
+                        },
+                        ColumnReader::Int64ColumnReader(ref mut r) => match dt {
+                            DataType::Time64(TimeUnit::Microsecond) => {
```

Comment:
There's aso `Time64(Nanoseconds)`

Reply:
Agree with @nevi-me , Time64(Nanoseconds) is missing.